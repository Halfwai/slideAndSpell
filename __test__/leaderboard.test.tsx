import React from 'react';
import { render, waitFor } from '@testing-library/react-native';

import LeaderBoard from '@/app/leaderboard';

beforeEach(() => {
    jest.useFakeTimers();
});

afterEach(() => {
    jest.clearAllMocks();
    jest.useRealTimers();
});

jest.mock('@/utils/supabaseFunctions', () => ({
    getLeaderboard: jest.fn(() =>
        Promise.resolve([
            {
                slides: 1,
                time_seconds: 1,
                display_name: 'test1',
                solution: [["a", "b", "c"], ["d", "e", "f"], ["g", "h", "i"]],
            },
        ])
    ),
}));

it(`LeaderBoard Component renders correctly`, async() => {
    const { toJSON } = render(
        <LeaderBoard/>
    );
    await waitFor(() => {
        expect(toJSON()).toMatchSnapshot();
    });
});