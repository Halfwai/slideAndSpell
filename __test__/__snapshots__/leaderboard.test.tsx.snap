// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderBoard Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#CBE6F7",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "height": "10%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "color": "black",
            "fontSize": 24,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Leaderboard
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#8BF0BC",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginBottom": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      MaterialCommunityIconsMock
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#8BF0BC",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginBottom": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Quicksand",
            },
            undefined,
          ]
        }
      >
        2025-03-15
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#8BF0BC",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginBottom": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      MaterialCommunityIconsMock
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderWidth": 1,
        "flex": 7,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          undefined,
        ]
      }
    >
      No entries yet
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1.5,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
</View>
`;
